### PR TITLE
[DEX-889] Add new release workflow

### PR DESCRIPTION
**Pull request by carine-bonnafous**
### Reason for change

Completes DEX-889

### Code changes

This adds : 
  * A CODEOWNERS file
  * pre-commit checks
  * The new commands `task release` and `task hotfix` to automate the creation of release pull requests, github release publication and communication, and backport PR creation 

### How to test

_As a reviewer, you are encouraged to test the PR locally._

### Checklist for authors and reviewers

- [ ] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [ ] The PR implements the changes asked in the referenced task / issue
- [ ] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [ ] The tests are relevant, and cover the corner/error cases, not only the happy path
- [ ] You understand the impact of this PR on existing code/features
- [ ] The changes include adequate logging and Datadog traces
- [ ] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -77,18 +77,71 @@ tasks:
...
+          Then, run `gh auth login` or set the GITHUB_TOKEN environment variable with your personal access token.
+
+  release:
+    deps:
+      - gh-cli
+    desc: Create a release pull request
+    cmds:
+      - gh workflow run release-pull-request.yml
+      - sleep 2
+      - cmd: echo "Release pull request created, check out https://github.com/alma/{{.REPOSITORY}}/pulls?q=is%3Aopen+is%3Apr+label%3Arelease"
```
Comment (gdraynz) [inline Taskfile.yml:119]: `{{ .REPOSITORY }}` is never set anywhere ?

Comment (carine-bonnafous) [reply Taskfile.yml:120]: Indeed🤦
Thanks, I pushed a fix

**Review by gdraynz — APPROVED** [review]

File: Taskfile.yml [hunk 2]
```diff
@@ -73,6 +73,26 @@ tasks:
...
+  pre-commit:
+    desc: Install pre-commit tool
+    internal: true
+    deps: [brew]
+    status:
+      - pre-commit --version
+    cmds:
+      - brew install pre-commit
+
+  install:pre-commit:
```
Comment (gdraynz) [inline Taskfile.yml:90]: Use this to have proper namespacing
```suggestion
  pre-commit:install
```

Comment (carine-bonnafous) [reply Taskfile.yml]: Thanks, done

**Review by gdraynz — APPROVED** [review]

**Pull request merged**